pact/Pact/Core/Info.hs

```diff
@@ -42,3 +44,15 @@ instance Pretty SpanInfo where
 combineSpan :: SpanInfo -> SpanInfo -> SpanInfo
 combineSpan (SpanInfo l1 c1 _ _) (SpanInfo _ _ l2 c2) =
   SpanInfo l1 c1 l2 c2
+
+data Located i a
+  = Located
+  { _locLocation :: i
+  , _locElem :: a }
+  deriving (Show, Functor, Foldable, Traversable)
+
+instance (Eq a) => Eq (Located i a) where
```

Comment:
> This seems problematic in case of multiple warnings across a file as the warnings are stored in a Set.

This makes sense, but I am unsure whether I want to accumulate as many warnings as possible. If we don't make this a set, users will have to pay gas for warnings.

Reply:
That makes sense. How about collecting error categories with an example position?

Reply:
We are simply going to collect them up to a bound